rasa/core/channels/slack.py

```diff
@@ -319,18 +333,30 @@ def blueprint(self, on_new_message):
                     return response.json(output.get("challenge"))
 
                 elif self._is_user_message(output):
-                    return await self.process_message(
-                        request,
-                        on_new_message,
-                        text=self._sanitize_user_message(
-                            output["event"]["text"], output["authed_users"]
-                        ),
-                        sender_id=output.get("event").get("user"),
-                    )
+                    channel = output["event"]["channel"]
+                    if (
+                        is_direct_message(output)
+                        or is_app_mention(output)
+                        or channel == self.slack_channel
+                    ):
+                        self.set_output_channel(channel)
```

Comment:
we can't change the channel of the slack input. it is shared between all the async incoming messages so you might change it while another request is also getting processed (this problem might not surface right now, if the asyncs & awaits are at the right )

Reply:
I think we should remove that object state on the input channel and directly pass it to the output channel

Reply:
@tmbo , I am using the metadata object to store this value as well as others related to a specific user request. I think we'll need to explore incorporating the new converation session feature from 1.6.0.